src/main/java/io/leonis/algieba/spatial/BoundaryPotentialField.java

```diff
@@ -0,0 +1,148 @@
+package io.leonis.algieba.spatial;
+
+import io.leonis.algieba.geometry.*;
+import java.util.function.UnaryOperator;
+import lombok.Value;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.factory.Nd4j;
+import org.nd4j.linalg.ops.transforms.Transforms;
+
+/**
+ * The Class BoundaryPotentialField.
+ *
+ * This class represents a {@link PotentialField} which bounds a rectangle such that the potential
+ * approximates zero for any point on the field and approximates infinity for any
+ * point outside the field.
+ *
+ * @author Rimon Oz
+ */
+@Value
+public class BoundaryPotentialField implements PotentialField {
+  private final INDArray origin = Nd4j.zeros(2, 1);
+  /**
+   * The width of the field in mm.
+   */
+  private final double width;
+  /**
+   * The length of the field in mm.
+   */
+  private final double length;
+  /**
+   * The displacement of the potential field from the edge.
+   */
+  private final double fieldDisplacement;
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public double getPotential(final INDArray positionVector) {
+    return this.getPotential(positionVector.getDouble(0, 0))
+        + this.getPotential(this.getWidth() - positionVector.getDouble(0, 0))
+        + this.getPotential(positionVector.getDouble(1, 0))
+        + this.getPotential(this.getLength() - positionVector.getDouble(1, 0));
+  }
+
+  /**
+   * See <a href="http://www.wolframalpha.com/input/?i=1%2F(1+%2B+e%5E(D+-+d))">this equation.</a>
```

Comment:
We need a way to include equations in the javadoc. although that's out of scope for now.